Libraries/PeriphDrivers/Source/SPI/spi_reva2.c

```diff
@@ -1476,23 +1526,179 @@ int MXC_SPI_RevA2_ControllerTransactionB(mxc_spi_reva_regs_t *spi, uint8_t *tx_b
                                          uint32_t tx_fr_len, uint8_t *rx_buffer, uint32_t rx_fr_len,
                                          uint8_t deassert, mxc_spi_target_t *target)
 {
```

Comment:
Maintenance overhead and code duplication for the `B` variants is a pretty major concern.  This could be handled better with an extra function argument

Reply:
I'll add create a function similar to the previous SPI API TransSetup function to minimize duplication.

Reply:
I reduced the maintenance overhead and code duplication for all the functions.